web/src/types.ts

```diff
@@ -192,3 +192,14 @@ export type UpdateMessageEvent = {
 // TODO/typescript: Move the User and Stream placeholder
 // types to their appropriate modules.
 export type User = Record<string, never>;
+
+export type GroupPermissionSetting = {
+    require_system_group: boolean;
+    allow_internet_group: boolean;
+    allow_owners_group: boolean;
+    allow_nobody_group: boolean;
+    allow_everyone_group: boolean;
+    default_group_name: string;
+    id_field_name: string;
+    default_for_system_groups?: string;
+};
```

Comment:
Moved this here, since it is used in `page_params` as well.

Reply:
Is `default_for_system_groups` optional or nullable? The type here has it optional, but the API types have it listed as nullable.

Reply:
Yeah, it is nullable. I got confused seeing that we do not pass the field in some cases but we set the default as `None`. Updated it.